### PR TITLE
enas nni support

### PR DESCRIPTION
**Pull request by countif**
enas nni support

### CODE REVIEW

File: nni/examples/trials/enas/config.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+authorName: Unknown
```
Comment (xuehui1991) [inline nni/examples/trials/enas/config.yml]: I think you need to write a README file about how to pull and install the nni dev-enas branch

File: nni/examples/trials/enas/config.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+experimentName: enas_macro
+trialConcurrency: 1
```
Comment (xuehui1991) [inline nni/examples/trials/enas/config.yml]: please test trialConcurrency = 2, too

File: nni/examples/trials/enas/config.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+maxExecDuration: 24h
+maxTrialNum: 1
```
Comment (xuehui1991) [inline nni/examples/trials/enas/config.yml]: please test maxTrialNum = 10

File: nni/examples/trials/enas/config.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#choice: local, remote
+trainingServicePlatform: local
+#choice: true, false
+useAnnotation: false
+multiPhase: true
+tuner:
+  codeDir: ~/nni/examples/tuners/enas
```
Comment (xuehui1991) [inline nni/examples/trials/enas/config.yml]: please mention the user to change this codeDir in README

File: nni/examples/trials/enas/data/ptb/process.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import cPickle as pickle
+import numpy as np
+
```
Comment (xuehui1991) [inline nni/examples/trials/enas/data/ptb/process.py]: only need two blank line

File: nni/examples/trials/enas/src/cifar10/data_utils.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  Returns:
+    images: np tensor of size [N, H, W, C]
+    labels: np tensor of size [N]
+  """
+  images, labels = [], []
+  for file_name in train_files:
+    print(file_name)
+    full_name = os.path.join(data_path, file_name)
+    with open(full_name,"rb") as finp:
+      data = pickle.load(finp,encoding='latin1')
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/data_utils.py:20]: just wonder why the encoding mode is 'latin1'

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+import tensorflow as tf
+import logging
+from src.utils import Logger
+from src.cifar10.data_utils import read_data
+from src.cifar10.general_child import GeneralChild
+from src.cifar10.micro_child import MicroChild
+from src.nni_child import ENASBaseTrial
+from  src.cifar10flags import *
+import nni
+def build_logger(log_name):
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py:22]: add two blank line.

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+    logger = logging.getLogger(log_name)
+    logger.setLevel(logging.DEBUG)
+    fh = logging.FileHandler(log_name+'.log')
+    fh.setLevel(logging.DEBUG)
+    logger.addHandler(fh)
+    return logger
+logger = build_logger("nni_child_cifar10")
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py:31]: same here

File: nni/examples/tuners/enas/nni_controller_cifar10.py [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+                log_string += " |g|={:<8.4f}".format(gn)
+                log_string += " acc={:<6.4f}".format(val_acc)
+                log_string += " bl={:<5.2f}".format(bl)
+                log_string += " child acc={:<5.2f}".format(valid_acc_arr[ct_step])
+                logger.debug(log_string)
+
+        return
+
+    def receive_trial_result(self, parameter_id, parameters, reward, trial_job_id):
+        logger.debug("epoch:\t"+str(self.epoch))
```
Comment (chicm-ms) [inline nni/examples/tuners/enas/nni_controller_cifar10.py:193]: @xuehui1991  It seems trial_job_id is not used, if trial_job_id is useless at all for enas, we can remove this parameter.

File: nni/examples/tuners/enas/nni_controller_cifar10.py [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+
+def main(_):
+    logger.debug("-" * 80)
+
+    SearchForMicro = False
+    if FLAGS.search_for == "micro":
+        SearchForMicro = True
+
+    logger.debug('Parse parameter done!!!!!!!!!!!!!!!!!!!!!!!!.')
+    logger.debug("Not here")
```
Comment (chicm-ms) [inline nni/examples/tuners/enas/nni_controller_cifar10.py]: Is this "not here" useful?

File: nni/examples/tuners/enas/nni_controller_cifar10.py [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+            out_file.write(str(number) + "\n")
+
+            for i in range(number):
+                arc = normal_arc[i]
+                self.writearcline(arc, out_file=out_file)
+                arc = reduce_arc[i]
+                self.writearcline(arc, out_file=out_file)
+
+
+def main(_):
```
Comment (chicm-ms) [inline nni/examples/tuners/enas/nni_controller_cifar10.py]: What is the purpose of this main function?

File: nni/examples/tuners/enas/nni_controller_cifar10.py (hunk 10 above)
Comment (countif) [reply nni/examples/tuners/enas/nni_controller_cifar10.py]: Just for debug .
I will remove it later

File: nni/examples/tuners/enas/nni_controller_cifar10.py (hunk 11 above)
Comment (countif) [reply nni/examples/tuners/enas/nni_controller_cifar10.py]: I will remove it later

File: nni/examples/tuners/enas/src/cifar10/data_utils.py [hunk 12]
```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+import cPickle as pickle
```
Comment (chicm-ms) [inline nni/examples/tuners/enas/src/cifar10/data_utils.py]: in python3, there is no cPickle, is this file used?

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 13]
```diff
@@ -0,0 +1,320 @@
...
+    child_totalsteps = (FLAGS.train_data_size + FLAGS.batch_size - 1) // FLAGS.batch_size
+    logger.debug("child total \t"+str(child_totalsteps))
+    epoch = 0
+
+    first_arc = None
+    if is_micro:
+        while True:
+            if epoch >= FLAGS.num_epochs:
+                break
+            # TODO nni.get_parameters()
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py]: remove TODO

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 14]
```diff
@@ -0,0 +1,320 @@
...
+            first_arc = (normal_arc[0], reduce_arc[0])
+            logger.debug("len\t" + str(len(normal_arc)))
+
+            actual_step, epoch = trial.run_cchild_one_micro(child_totalsteps, normal_arc, reduce_arc)
+
+            logger.debug("epoch:\t" + str(epoch)+"actual_step:\t"+str(actual_step))
+            valid_acc_arr = trial.get_child_arc_micro(controller_total_steps, normal_arc, reduce_arc)
+            logger.debug("Get rewards Done!\n")
+
+            # TODO: nni.send_final_result()
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py]: remove TODO

File: nni/examples/tuners/enas/src/cifar10/data_utils.py (hunk 12 above)
Comment (countif) [reply nni/examples/tuners/enas/src/cifar10/data_utils.py]: Original code is python2, I will remove it.

File: nni/examples/tuners/enas/src/utils.py [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+      lr_init, tf.maximum(train_step - lr_dec_start, 0), lr_dec_every,
+      lr_dec_rate, staircase=True)
+    if lr_dec_min is not None:
+      learning_rate = tf.maximum(learning_rate, lr_dec_min)
+
+  if lr_warmup_val is not None:
+    learning_rate = tf.cond(tf.less(train_step, lr_warmup_steps),
+                            lambda: lr_warmup_val, lambda: learning_rate)
+
+  # if get_grad_norms:
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/utils.py]: remove

File: nni/examples/tuners/enas/src/ptb/ptb_enas_controller.py [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+    sample_entropy = tf.concat(sample_entropy, axis=0)
+    self.sample_entropy = tf.reduce_sum(sample_entropy)
+
+    self.all_h = all_h
+
+  def build_trainer(self):
+    # actor
+    self.valid_loss = tf.placeholder(dtype=tf.float32, shape=[])
+
+    #self.valid_loss = tf.to_float(child_model.rl_loss)
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/ptb_enas_controller.py:180]: remove

File: nni/examples/tuners/enas/src/ptb/ptb_enas_controller.py [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+    all_h_w = []
+
+    # sampler ops
+    inputs = self.g_emb
+    prev_c, prev_h = [], []
+    for _ in xrange(self.lstm_num_layers):
+      prev_c.append(tf.zeros([1, self.lstm_size], dtype=tf.float32))
+      prev_h.append(tf.zeros([1, self.lstm_size], dtype=tf.float32))
+
+    # used = tf.zeros([self.rhn_depth, 2], dtype=tf.int32)
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/ptb_enas_controller.py:108]: remove

File: nni/examples/tuners/enas/src/nni_controller.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+    def get_csvaa(self, child_totalsteps):
+        child_arc = []
+        for _ in range(0, child_totalsteps):
+            arc = self.sess.run(self.controller_model.sample_arc)
+            child_arc.append(arc)
+        return child_arc
+
+
+    # TODO: nni.send_final_result()
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/nni_controller.py]: remove

File: nni/examples/tuners/enas/src/nni_controller.py [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+from src.cifar10.micro_controller import MicroController
+import nni
+from nni.multi_phase.multi_phase_tuner import MultiPhaseTuner
+
+class ENASBaseTuner(MultiPhaseTuner):
+
+    def __init__(self):
+        return
+
+    def get_csvaa(self, child_totalsteps):
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/nni_controller.py]: what's the meaning of "csvaa"???

File: nni/examples/tuners/enas/src/nni_controller.py [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+            fcntl.flock(out_file, fcntl.LOCK_EX)
+            number = len(child_arc)
+            out_file.write(str(number) + "\n")
+            for i in range(len(child_arc)):
+                arc = child_arc[i]
+                self.writearcline(arc, out_file)
+        return
+
+
+    def writearcline(self, arc, out_file):
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/nni_controller.py]: write_arc_line

File: nni/examples/tuners/enas/src/nni_controller.py [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+        line = str(arc[0])
+        arc_len = len(arc)
+        for idx in range(1, arc_len, 1):
+            line = line + " " + str(arc[idx])
+        out_file.write(line + "\n")
+
+
+    #TODO nni.get_parameters()
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/nni_controller.py]: remove unused code

File: nni/examples/trials/enas/config.yml [hunk 22]
```diff
@@ -0,0 +1,21 @@
+authorName: Unknown
+experimentName: enas_macro
```
Comment (xuehui1991) [inline nni/examples/trials/enas/config.yml]: Could you add some unit-test to make some basic function work ok????

File: nni/examples/trials/enas/src/cifar10/data_utils.py (hunk 6 above)
Comment (countif) [reply nni/examples/trials/enas/src/cifar10/data_utils.py:20]: I get this from google.

File: nni/examples/trials/enas/src/nni_child.py [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+import logging
+from src.utils import Logger
+from src.utils import DEFINE_boolean
+from src.utils import DEFINE_float
+from src.utils import DEFINE_integer
+from src.utils import DEFINE_string
+from src.cifar10.data_utils import read_data
+from src.cifar10.general_child import GeneralChild
+from src.cifar10.micro_child import MicroChild
+
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/nni_child.py:21]: remove 1 blank line

File: nni/examples/trials/enas/src/nni_child.py [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+
+
+class ENASBaseTrial(object):
+
+    def __init__(self):
+        return
+
+
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/nni_child.py:28]: remove 1 blank line

File: nni/examples/trials/enas/src/nni_child.py [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+        valid_acc_arr = []
+        for idx in range(0, controller_total_steps):
+            cur_valid_acc = self.sess.run\
+                (self.child_model.cur_valid_acc,
+                 feed_dict={self.child_model.sample_arc: child_arc[idx]})
+            valid_acc_arr.append(cur_valid_acc)
+        return valid_acc_arr
+
+
+    # TODO: nni.send_final_result()
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/nni_child.py]: remove

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 26]
```diff
@@ -0,0 +1,320 @@
...
+    logger.debug("child total \t"+str(child_totalsteps))
+    epoch = 0
+
+    first_arc = None
+    if is_micro:
+        while True:
+            if epoch >= FLAGS.num_epochs:
+                break
+            # TODO nni.get_parameters()
+            normal_arc, reduce_arc = trial.receive_micro_arc(epoch)
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py]: change api

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 27]
```diff
@@ -0,0 +1,320 @@
...
+            logger.debug("len\t" + str(len(normal_arc)))
+
+            actual_step, epoch = trial.run_cchild_one_micro(child_totalsteps, normal_arc, reduce_arc)
+
+            logger.debug("epoch:\t" + str(epoch)+"actual_step:\t"+str(actual_step))
+            valid_acc_arr = trial.get_child_arc_micro(controller_total_steps, normal_arc, reduce_arc)
+            logger.debug("Get rewards Done!\n")
+
+            # TODO: nni.send_final_result()
+            trial.send_reward(epoch, valid_acc_arr)
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py]: chaneg api

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py [hunk 28]
```diff
@@ -0,0 +1,320 @@
...
+            trial.start_eval_micro(first_arc=first_arc)
+    else:
+        while True:
+            if epoch >= FLAGS.num_epochs:
+                break
+            # TODO nni.get_parameters()
+            logger.debug("get paramters")
+            child_arc = nni.get_parameters()
+            child_arc = trial.parset_child_arch(child_arc)
+            #child_arc = trial.receive_macro_arc(epoch)
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py]: remove unused code

File: nni/examples/trials/enas/src/nni_child.py [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+    def send_reward(self, epoch, rewards):
+        with open(self.child_prefix + str(epoch) + ".txt", "w") as out_file:
+            fcntl.flock(out_file, fcntl.LOCK_EX)
+            number = len(rewards)
+            out_file.write(str(number) + "\n")
+            for i in range(0, number):
+                out_file.write(str(rewards[i]) + "\n")
+
+
+    def receive_line(self, line):
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/nni_child.py]: remove unused file

File: nni/examples/trials/enas/src/nni_child.py [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
+        tokens = line.split()
+        arc = []
+        for j in range(0, len(tokens)):
+            arc.append(int(tokens[j]))
+        arc = np.array(arc)
+        return arc
+
+
+    # TODO nni.get_parameters()
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/nni_child.py]: remove unused code

File: nni/examples/trials/enas/src/ptb/nni_child_ptb.py [hunk 31]
```diff
@@ -0,0 +1,230 @@
...
+from src.ptbflags import *
+
+def build_logger(log_name):
+    logger = logging.getLogger(log_name)
+    logger.setLevel(logging.DEBUG)
+    fh = logging.FileHandler(log_name+'.log')
+    fh.setLevel(logging.DEBUG)
+    logger.addHandler(fh)
+    return logger
+logger = build_logger("nni_child_ptb")
```
Comment (xuehui1991) [inline nni/examples/trials/enas/src/ptb/nni_child_ptb.py:32]: add two blank line..

File: nni/examples/tuners/enas/src/ptb/main.py [hunk 32]
```diff
@@ -0,0 +1,250 @@
...
+    "num_train_batches": child_model.num_train_batches,
+    "eval_every": child_model.num_train_batches * FLAGS.eval_every_epochs,
+    "eval_func": child_model.eval_once,
+  }
+
+
+  return controller_ops,child_ops
+
+
+
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/main.py]: remove blank line...

File: nni/examples/tuners/enas/src/ptb/main.py [hunk 33]
```diff
@@ -0,0 +1,250 @@
...
+def getcontrollerlog(sess,controller_ops):
+  print("Here are 10 architectures")
+  for _ in xrange(10):
+    arc, rw = sess.run([
+      controller_ops["sample_arc"],
+      controller_ops["reward"],
+    ])
+    print("{} rw={:<.3f}".format(np.reshape(arc, [-1]), rw))
+  return
+def getarchlog(child_ops,sess,epoch,best_valid_ppl):
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/main.py]: add blank line

File: nni/examples/tuners/enas/src/ptb/main.py [hunk 34]
```diff
@@ -0,0 +1,250 @@
...
+
+
+  return controller_ops,child_ops
+
+
+
+
+
+
+def getcontrollerlog(sess,controller_ops):
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/main.py]: get_controller_log

File: nni/examples/tuners/enas/src/ptb/main.py [hunk 35]
```diff
@@ -0,0 +1,250 @@
...
+  print("Here are 10 architectures")
+  for _ in xrange(10):
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/main.py]: set 10 as a parameter user could pass in, default value is 10

File: nni/examples/tuners/enas/src/ptb/main.py [hunk 36]
```diff
@@ -0,0 +1,250 @@
...
+          sys.exit(0)
+
+        num_batches = 0
+        total_tr_ppl = 0
+        best_valid_ppl = 67.00
+        total_steps =  FLAGS.controller_train_every * child_ops["num_train_batches"]
+        controller_total_steps = FLAGS.controller_train_steps * FLAGS.controller_num_aggregate
+
+        while True:
+
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/ptb/main.py]: remove blank line

File: nni/examples/tuners/enas/src/cifar10flags.py [hunk 37]
```diff
@@ -0,0 +1,46 @@
+import tensorflow as tf
+from src.utils import DEFINE_boolean
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/src/cifar10flags.py]: could you change the name "cifar10flags.py" to "cifar10_flags.py"?

File: nni/examples/tuners/enas/script/microcontroller.sh [hunk 38]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/script/microcontroller.sh]: change the name to "micro_controller"？

File: nni/examples/tuners/enas/script/macrocontroller.sh [hunk 39]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/script/macrocontroller.sh]: change the name to "macro_controller"?

File: nni/examples/tuners/enas/nni_controller_cifar10.py [hunk 40]
```diff
@@ -0,0 +1,207 @@
...
+    def receive_trial_result(self, parameter_id, parameters, reward, trial_job_id):
+        logger.debug("epoch:\t"+str(self.epoch))
+        logger.debug(parameter_id)
+        logger.debug(reward)
+        valid_acc_arr = reward
+        self.controller_one_step(self.epoch, valid_acc_arr)
+        return
+
+
+    def update_search_space(self, data):
```
Comment (xuehui1991) [inline nni/examples/tuners/enas/nni_controller_cifar10.py:201]: you could raise a RuntimeError here.   ENASTuner do not support update search space now

File: nni/examples/tuners/enas/nni_controller_cifar10.py (hunk 9 above)
Comment (xuehui1991) [reply nni/examples/tuners/enas/nni_controller_cifar10.py:193]: for the single machine version, the trial_jon_id is useless, but for multi-machine version, we need it.

File: README.md [hunk 41]
```diff
@@ -1 +1,13 @@
-# enas_nni
\ No newline at end of file
+# enas_nni
+This code is for running enas code on nni system.
+link:https://github.com/Microsoft/nni
+link:https://github.com/melodyguan/enas
+
+Install:
+  1 Need to install nni system first.
+  link:https://github.com/Microsoft/nni You need to choose the dev-enas branch.
```
Comment (xuehui1991) [inline README.md]: pip3 install -v --user git+https://github.com/Microsoft/nni.git@dev-enas
source ~/.bashrc

please use markdown.

File: README.md [hunk 42]
```diff
@@ -1 +1,13 @@
...
+  2 Modify the codeDir of  enas_nni/nni/examples/trials/enas/config.yml.
+  3 Download the cifar10 data at https://www.cs.toronto.edu/~kriz/cifar.html (python version). You need to put data at  enas_nni/nni/examples/trials/enas/data and rename the cifar-10-batches-py as cifar10 .
```
Comment (xuehui1991) [inline README.md]: can you write a wget script here?

File: nni/examples/tuners/enas/script/macrocontroller.sh (hunk 39 above)
Comment (countif) [reply nni/examples/tuners/enas/script/macrocontroller.sh]: It seems we do not need this now.

File: README.md [hunk 43]
```diff
@@ -1 +1,13 @@
...
+This code is for running enas code on nni system.
+link:https://github.com/Microsoft/nni
+link:https://github.com/melodyguan/enas
+
+Install:
+  1 Need to install nni system first.
+  link:https://github.com/Microsoft/nni You need to choose the dev-enas branch.
+  2 Modify the codeDir of  enas_nni/nni/examples/trials/enas/config.yml.
+  3 Download the cifar10 data at https://www.cs.toronto.edu/~kriz/cifar.html (python version). You need to put data at  enas_nni/nni/examples/trials/enas/data and rename the cifar-10-batches-py as cifar10 . 
+  4 Run the command: nnictl create --config ~/chicm/nni/examples/trials/enas/config.yml (It depends on where you put the config on)
```
Comment (xuehui1991) [inline README.md]: remove the path /chicm/. It's not good for user.

File: nni/examples/tuners/enas/nni_controller_cifar10.py (hunk 40 above)
Comment (countif) [reply nni/examples/tuners/enas/nni_controller_cifar10.py:201]: I use pass . follow the customer tuner, And it runs well.

File: README.md (hunk 43 above)
Comment (xuehui1991) [inline README.md]: how to run other mode?
Can you add the paper and tell the user what's are you doing here?

File: nni/examples/tuners/enas/nni_controller_cifar10.py (hunk 9 above)
Comment (xuehui1991) [reply nni/examples/tuners/enas/nni_controller_cifar10.py:193]: @chicm-ms

File: README.md (hunk 43 above)
Comment (countif) [reply README.md]: Haven't support other mode.
I will add the paper here

File: nni/examples/tuners/enas/src/nni_controller.py (hunk 19 above)
Comment (countif) [reply nni/examples/tuners/enas/src/nni_controller.py]: I will choose a better name.

File: nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py (hunk 27 above)
Comment (countif) [reply nni/examples/trials/enas/src/cifar10/nni_child_cifar10.py]: This is for micro . I will change it later.

File: README.md [hunk 44]
```diff
@@ -1 +1,42 @@
-# enas_nni
\ No newline at end of file
+# enas_nni
+>This code is for running enas code on nni system.
```
Comment (xuehui1991) [inline README.md]: nni is a package instead of system...

**Review by xuehui1991 — APPROVED** [review]

**Pull request merged**